app/controllers/service_controller.rb

```diff
@@ -150,6 +157,22 @@ def service_form_fields
     }
   end
 
+  def generic_object
+    return unless init_show_variables
+
+    id = params[:show] || params[:x_show]
+    if id.present?
```

Comment:
This whole block is surely repeated in places. I am quite sure there's a method or even methods that do this.